x-pack/plugins/enterprise_search/public/applications/enterprise_search_content/components/connector_detail/attach_index_box.tsx

```diff
@@ -140,6 +143,10 @@ export const AttachIndexBox: React.FC<AttachIndexBoxProps> = ({ connector }) =>
     }
   }, [query]);
 
+  useEffect(() => {
```

Comment:
nitpick: don't really need an effect for this. You can just do `const sanitizedName = formatApiName(connector.name)` higher up in this component. Can wrap it in a `useMemo` if you want.